### ISSUE TITLE
[NPC][Quest][WotlK] Injured Goblin Miner - Bitter Departure ,missing spawns

### ISSUE THREAD

**Issue by Teridan-dal-wow** [dalaranwow/dalaran-wow#4303]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: 
Part 1: There should be 3 npcs Injured Goblin Miner http://wotlk.openwow.com/npc=29434 offering the same quests, however these 3 miners are spawned on diffrent places in the cave so they have abit diffrent escort path (check video below).
Note: Currently we have 1 miner spawned here with working quests and path:
![wowscrnshot_012417_103653](https://cloud.githubusercontent.com/assets/6797611/22239883/0f9f0b1e-e221-11e6-9db1-818f3b294da4.jpg)

The spots with the missing miners are:
![wowscrnshot_012417_103857](https://cloud.githubusercontent.com/assets/6797611/22239981/824d5e04-e221-11e6-9433-f5da4cec7ab6.jpg)
![wowscrnshot_012417_104009](https://cloud.githubusercontent.com/assets/6797611/22239983/85246230-e221-11e6-81b8-502229757271.jpg)

**Expected behaviour**: Video from retail legion from 24.01.2017:
Part 1-https://www.youtube.com/watch?v=2be1QkWWEj0
Part 2-https://www.youtube.com/watch?v=kgPxndqu9gs
Speech from part 2:
[10:50:21] Injured Goblin Miner says: Let me know when you're ready. I'd prefer sooner than later... what with the slowly dying from poison and all.
[10:51:28] Injured Goblin Miner says: I'm going to bring the venom sac to Ricket... and then... you know... collapse. Thank you for helping me!

Comment 1: All 3 miners share the same quests, so if you complete 1 of the 2 quests in their chain, it should be updated for all of them, meaning you shouldnt be able to complete the escort quests 3 times.
Comment 2: Both Crystalweb Weaver and Snowblind Digger are hostile towards the Miners while the escort quest is started, when its not they should ignore the them.

Related report #4294

**Comment by Teridan-dal-wow** [comment]
@Rushor for this report, lets say all 3 of these goblin are missing, where do i event start to script that quest?

**Comment by Rushor** [comment]
you need to improve the corescript then npc_injured_goblin. i will do it for ya since you need a lot of time to learn about all this c++ stuff

**Comment by Rushor** [comment]
in this case we have to add a guid check for the creature, depending on the guid it should start a different path. we could also convert the whole script to db if you want. i will post the fix later

**Comment by Teridan-dal-wow** [comment]
Yes, i would like to stick with db, everything else is too complicated.

**Comment by Rushor** [comment]
@Teridan-dal-wow could you create the waypoints for it? maybe add it with the format of waypoint_data i will convert it later then. the db script is finsihed so far, all i need are the waypoint of the three different escortpaths now

**Comment by Teridan-dal-wow** [comment]
Starting.

**Comment by Teridan-dal-wow** [comment]
``` sql
-- Pathing for Injured Goblin Miner Entry: 29434 for part 1 video
SET @NPC := XXXXXX;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6675.684,`position_y`=-1133.351,`position_z`=397.0984 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6675.684,-1133.351,397.0984,0,0,1,0,100,0), -- 10:27:29
(@PATH,2,6685.072,-1098.411,396.1642,0,0,1,0,100,0), -- 10:27:31
(@PATH,3,6681.2,-1080.326,399.9615,0,0,1,0,100,0), -- 10:27:36
(@PATH,4,6679.967,-1046.549,409.238,0,0,1,0,100,0), -- 10:27:40
(@PATH,5,6674.312,-1034.813,412.6546,0,0,1,0,100,0), -- 10:27:45
(@PATH,6,6656.498,-1017.064,415.3774,0,0,1,0,100,0), -- 10:27:47
(@PATH,7,6633.972,-1014.241,424.0492,0,0,1,0,100,0), -- 10:27:51
(@PATH,8,6595.558,-1017.468,429.5381,0,0,1,0,100,0), -- 10:27:54
(@PATH,9,6564.813,-1027.838,433.8102,0,0,1,0,100,0), -- 10:28:00
(@PATH,10,6524.622,-1039.275,433.2169,0,0,1,0,100,0), -- 10:28:06
(@PATH,11,6499.231,-1042.115,432.5026,0,0,1,0,100,0), -- 10:28:12
(@PATH,12,6469.883,-1031.083,434.3615,0,0,1,0,100,0), -- 10:28:16
(@PATH,13,6441.251,-1026.213,430.2799,0,0,1,0,100,0), -- 10:28:22
(@PATH,14,6437.511,-1025.804,429.1273,0.122173,0,1,0,100,0), -- 10:28:27
(@PATH,15,6374.633,-1040.804,420.2216,0,0,1,0,100,0); -- 10:28:32
-- Pathing for Injured Goblin Miner Entry: 29434 for part 2 video
SET @NPC := XXXXXX;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6664.493,`position_y`=-1106.511,`position_z`=423.9608 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6664.493,-1106.511,423.9608,0,0,1,0,100,0), -- 10:44:02
(@PATH,2,6674.33,-1120.81,424.4858,0,0,1,0,100,0), -- 10:44:06
(@PATH,3,6669.01,-1129.653,423.9612,0,0,1,0,100,0), -- 10:44:10
(@PATH,4,6634.209,-1128.281,417.0604,0,0,1,0,100,0), -- 10:44:12
(@PATH,5,6620.618,-1119.293,412.5555,0,0,1,0,100,0), -- 10:44:17
(@PATH,6,6617.512,-1092.118,411.8932,0,0,1,0,100,0), -- 10:44:21
(@PATH,7,6620.42,-1077.044,415.567,0,0,1,0,100,0), -- 10:44:29
(@PATH,8,6628.931,-1050.362,415.3565,0,0,1,0,100,0), -- 10:44:30
(@PATH,9,6623.998,-1030.46,423.5757,0,0,1,0,100,0), -- 10:44:34
(@PATH,10,6613.661,-1021.642,427.4341,0,0,1,0,100,0), -- 10:44:38
(@PATH,11,6604.476,-1021.027,428.391,0,0,1,0,100,0), -- 10:44:40
(@PATH,12,6603.476,-1023.777,428.391,0,0,1,0,100,0), -- 10:44:42
(@PATH,13,6565.581,-1026.838,433.7635,0,0,1,0,100,0), -- 10:44:45
(@PATH,14,6525.857,-1036.689,433.0396,0,0,1,0,100,0), -- 10:44:57
(@PATH,15,6497.649,-1040.301,432.9883,0,0,1,0,100,0), -- 10:44:58
(@PATH,16,6483.933,-1039.602,433.0212,0,0,1,0,100,0), -- 10:45:03
(@PATH,17,6440.578,-1042.148,428.8166,0,0,1,0,100,0), -- 10:45:06
(@PATH,18,6373.59,-1042.169,420.1453,0,0,1,0,100,0); -- 10:45:18

**Comment by Rushor** [comment]
nice ty

**Comment by Rushor** [comment]
hm there is another issue with the gossips - you should only be able to see the gossip option if have accepted the quest - not always

**Comment by Teridan-dal-wow** [comment]
Note: The last wp in each case (posted above) is the place where goblins says the 2-nd speech, however after that the goblin runs towards K3 are despawns after some distance (with increased movement speed), You can see this element at part 1 video 1:31s (i think that part wasn't working on live).

**Comment by Rushor** [comment]
> with increased movement speed

for this part we have to add a new sai action in core too

**Comment by Rushor** [comment]
alrght the converted part for db:

```sql
-- Bitter Departure
UPDATE `creature_template` SET `ScriptName`='' WHERE  `entry`=29434;
DELETE FROM `script_waypoint` WHERE  `entry`=29434;

-- Injured Goblin Miner SAI
SET @ENTRY := 29434;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
-- gossip select part
(@ENTRY,0,0,1,62,0,100,0,9859,0,0,0,2,113,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Set Faction 113"),
(@ENTRY,0,1,2,61,0,100,0,0,0,0,0,53,1,29434,0,0,0,1,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Start Waypoint"),
(@ENTRY,0,2,3,61,0,100,0,0,0,0,0,64,1,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Store Targetlist"),
(@ENTRY,0,3,4,61,0,100,0,0,0,0,0,83,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Npc Flags Gossip & Questgiver"),
-- remove it on gossip select too if player starts the quest after serverdown
(@ENTRY,0,4,5,61,0,100,0,0,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Aura 'Feign Death'"),
(@ENTRY,0,5,6,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Flag Standstate Dead"),
(@ENTRY,0,7,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Close Gossip"),
-- quest start part
(@ENTRY,0,8,9,19,0,100,0,12832,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Aura 'Feign Death'"),
(@ENTRY,0,9,10,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Flag Standstate Dead"),
(@ENTRY,0,10,11,61,0,100,0,0,0,0,0,1,0,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Say Line 0"),
(@ENTRY,0,11,0,61,0,100,0,0,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Set Active On"),
-- creature spawn part
(@ENTRY,0,12,13,37,0,100,0,0,0,0,0,2,35,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Faction 35"),
(@ENTRY,0,13,14,61,0,100,0,0,0,0,0,75,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Aura 'Feign Death'"),
(@ENTRY,0,14,15,61,0,100,0,0,0,0,0,90,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Flag Standstate Dead"),
(@ENTRY,0,15,0,61,0,100,0,0,0,0,0,82,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Npc Flags Gossip & Questgiver"),
(@ENTRY,0,16,0,6,0,100,0,0,0,0,0,6,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Just Died - Fail Quest 'Bitter Departure'"),
(@ENTRY,0,17,0,40,0,100,0,27,29434,0,0,80,@ENTRY*100+00,2,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Waypoint 27 Reached - Run Script");

-- Actionlist SAI
SET @ENTRY := 2943400;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,66,0,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Set Orientation <unsupported target type>"),
(@ENTRY,9,1,0,0,0,100,0,1000,1000,0,0,1,1,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Say Line 1"),
(@ENTRY,9,2,0,0,0,100,0,0,0,0,0,15,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Quest Credit 'Bitter Departure'"),
(@ENTRY,9,3,0,0,0,100,0,5000,5000,0,0,69,0,0,0,0,0,0,8,0,0,0,6312.16,-1038.56,414.128,3.2475,"Injured Goblin Miner - On Script - Move To Position"),
(@ENTRY,9,4,0,0,0,100,0,5000,5000,0,0,41,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Despawn Instant");


-- gossip option conditions
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=15 AND `SourceGroup`=9859 AND `SourceEntry`=0;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(15,9859,0,0,0,9,0,12832,0,0,0,0,0,"","Show Gossip Injured Goblin Miner only if player has quest 'Bitter Departure' Taken");

DELETE FROM `waypoints` WHERE `entry`=29434;
INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES 
(29434, 1, 6645.47, -1263.66, 396.938, 'Injured Goblin Miner'),
(29434, 2, 6662.06, -1274.13, 397.319, 'Injured Goblin Miner'),
(29434, 3, 6666.29, -1254.59, 396.11, 'Injured Goblin Miner'),
(29434, 4, 6669.33, -1240.06, 397.789, 'Injured Goblin Miner'),
(29434, 5, 6654.31, -1205.01, 399.823, 'Injured Goblin Miner'),
(29434, 6, 6657.03, -1170.15, 398.755, 'Injured Goblin Miner'),
(29434, 7, 6667.33, -1146.28, 399.014, 'Injured Goblin Miner'),
(29434, 8, 6670.9, -1138.68, 397.125, 'Injured Goblin Miner'),
(29434, 9, 6685.09, -1103.95, 396.828, 'Injured Goblin Miner'),
(29434, 10, 6688.71, -1094.32, 395.019, 'Injured Goblin Miner'),
(29434, 11, 6684.75, -1084.31, 396.916, 'Injured Goblin Miner'),
(29434, 12, 6679.3, -1074.4, 404.284, 'Injured Goblin Miner'),
(29434, 13, 6691.91, -1051.96, 407.983, 'Injured Goblin Miner'),
(29434, 14, 6696.28, -1016.54, 414.823, 'Injured Goblin Miner'),
(29434, 15, 6675.32, -1005.06, 414.844, 'Injured Goblin Miner'),
(29434, 16, 6661.01, -1007.41, 414.946, 'Injured Goblin Miner'),
(29434, 17, 6643.75, -1010.24, 420.508, 'Injured Goblin Miner'),
(29434, 18, 6620.76, -1012.59, 424.378, 'Injured Goblin Miner'),
(29434, 19, 6610.33, -996.992, 428.116, 'Injured Goblin Miner'),
(29434, 20, 6581.3, -1005.9, 434.257, 'Injured Goblin Miner'),
(29434, 21, 6570.74, -1013, 434.076, 'Injured Goblin Miner'),
(29434, 22, 6551.68, -1012.38, 433.218, 'Injured Goblin Miner'),
(29434, 23, 6530.83, -1024.99, 433.04, 'Injured Goblin Miner'),
(29434, 24, 6510.05, -1031.23, 435.824, 'Injured Goblin Miner'),
(29434, 25, 6491.5, -1032.46, 434.226, 'Injured Goblin Miner'),
(29434, 26, 6475.58, -1023.13, 434.759, 'Injured Goblin Miner'),
(29434, 27, 6451.81, -1025.43, 431.502, 'Injured Goblin Miner');
```

now we need to add that script to the 3 indivisual guids with different paths

**Comment by Rushor** [comment]
alright @Teridan-dal-wow 

added everything so far except the increased speed if the creature should walk to k3 (i will create a new issue for this with the coresupport then)

would you like to test this? :)

```sql
-- Bitter Departure
UPDATE `creature_template` SET `ScriptName`='' WHERE  `entry`=29434;
DELETE FROM `script_waypoint` WHERE  `entry`=29434;

SET @CGUID := 5004920;
DELETE FROM `creature` WHERE `guid` BETWEEN @CGUID+0 AND @CGUID+1;
INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`,  `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `VerifiedBuild`) VALUES
(@CGUID+0, 29434, 571, 0, 0, 1, 1, 0,  0, 6646.888, -1110.959, 427.3377, 3.228859, 25, 0, 0, 0, 0, 0, 0, 0, 0, 23360), -- Injured Goblin Miner (Area: -Unknown-) (Auras: 51329 - Feign Death)
(@CGUID+1, 29434, 571, 0, 0, 1, 1, 0,  0, 6690.494, -1167.411, 398.8149, 3.769911, 25, 0, 0, 0, 0, 0, 0, 0, 0, 23360); -- Injured Goblin Miner (Area: -Unknown-) (Auras: 51329 - Feign Death)

-- Injured Goblin Miner SAI
SET @ENTRY := -202337;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=29434;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
-- gossip select part
(@ENTRY,0,0,1,62,0,100,0,9859,0,0,0,2,113,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Set Faction 113"),
(@ENTRY,0,1,2,61,0,100,0,0,0,0,0,53,1,2943400,0,0,0,1,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Start Waypoint"),
(@ENTRY,0,2,3,61,0,100,0,0,0,0,0,64,1,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Store Targetlist"),
(@ENTRY,0,3,4,61,0,100,0,0,0,0,0,83,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Npc Flags Gossip & Questgiver"),
-- remove it on gossip select too if player starts the quest after serverdown
(@ENTRY,0,4,5,61,0,100,0,0,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Aura 'Feign Death'"),
(@ENTRY,0,5,6,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Flag Standstate Dead"),
(@ENTRY,0,7,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Close Gossip"),
-- quest start part
(@ENTRY,0,8,9,19,0,100,0,12832,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Aura 'Feign Death'"),
(@ENTRY,0,9,10,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Flag Standstate Dead"),
(@ENTRY,0,10,11,61,0,100,0,0,0,0,0,1,0,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Say Line 0"),
(@ENTRY,0,11,0,61,0,100,0,0,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Set Active On"),
-- creature spawn part
(@ENTRY,0,12,13,37,0,100,0,0,0,0,0,2,35,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Faction 35"),
(@ENTRY,0,13,14,61,0,100,0,0,0,0,0,75,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Aura 'Feign Death'"),
(@ENTRY,0,14,15,61,0,100,0,0,0,0,0,90,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Flag Standstate Dead"),
(@ENTRY,0,15,0,61,0,100,0,0,0,0,0,82,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Npc Flags Gossip & Questgiver"),
(@ENTRY,0,16,0,6,0,100,0,0,0,0,0,6,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Just Died - Fail Quest 'Bitter Departure'"),
(@ENTRY,0,17,0,40,0,100,0,27,2943400,0,0,80,2943400,2,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Waypoint 27 Reached - Run Script");

-- Injured Goblin Miner SAI
SET @ENTRY := -5004920;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=29434;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
-- gossip select part
(@ENTRY,0,0,1,62,0,100,0,9859,0,0,0,2,113,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Set Faction 113"),
(@ENTRY,0,1,2,61,0,100,0,0,0,0,0,53,1,2943402,0,0,0,1,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Start Waypoint"),
(@ENTRY,0,2,3,61,0,100,0,0,0,0,0,64,1,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Store Targetlist"),
(@ENTRY,0,3,4,61,0,100,0,0,0,0,0,83,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Npc Flags Gossip & Questgiver"),
-- remove it on gossip select too if player starts the quest after serverdown
(@ENTRY,0,4,5,61,0,100,0,0,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Aura 'Feign Death'"),
(@ENTRY,0,5,6,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Flag Standstate Dead"),
(@ENTRY,0,7,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Close Gossip"),
-- quest start part
(@ENTRY,0,8,9,19,0,100,0,12832,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Aura 'Feign Death'"),
(@ENTRY,0,9,10,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Flag Standstate Dead"),
(@ENTRY,0,10,11,61,0,100,0,0,0,0,0,1,0,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Say Line 0"),
(@ENTRY,0,11,0,61,0,100,0,0,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Set Active On"),
-- creature spawn part
(@ENTRY,0,12,13,37,0,100,0,0,0,0,0,2,35,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Faction 35"),
(@ENTRY,0,13,14,61,0,100,0,0,0,0,0,75,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Aura 'Feign Death'"),
(@ENTRY,0,14,15,61,0,100,0,0,0,0,0,90,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Flag Standstate Dead"),
(@ENTRY,0,15,0,61,0,100,0,0,0,0,0,82,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Npc Flags Gossip & Questgiver"),
(@ENTRY,0,16,0,6,0,100,0,0,0,0,0,6,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Just Died - Fail Quest 'Bitter Departure'"),
(@ENTRY,0,17,0,40,0,100,0,17,2943402,0,0,80,2943400,2,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Waypoint 17 Reached - Run Script");

-- Injured Goblin Miner SAI
SET @ENTRY := -5004921;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=29434;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
-- gossip select part
(@ENTRY,0,0,1,62,0,100,0,9859,0,0,0,2,113,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Set Faction 113"),
(@ENTRY,0,1,2,61,0,100,0,0,0,0,0,53,1,2943401,0,0,0,1,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Start Waypoint"),
(@ENTRY,0,2,3,61,0,100,0,0,0,0,0,64,1,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Store Targetlist"),
(@ENTRY,0,3,4,61,0,100,0,0,0,0,0,83,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Npc Flags Gossip & Questgiver"),
-- remove it on gossip select too if player starts the quest after serverdown
(@ENTRY,0,4,5,61,0,100,0,0,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Aura 'Feign Death'"),
(@ENTRY,0,5,6,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Flag Standstate Dead"),
(@ENTRY,0,7,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Close Gossip"),
-- quest start part
(@ENTRY,0,8,9,19,0,100,0,12832,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Aura 'Feign Death'"),
(@ENTRY,0,9,10,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Flag Standstate Dead"),
(@ENTRY,0,10,11,61,0,100,0,0,0,0,0,1,0,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Say Line 0"),
(@ENTRY,0,11,0,61,0,100,0,0,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Set Active On"),
-- creature spawn part
(@ENTRY,0,12,13,37,0,100,0,0,0,0,0,2,35,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Faction 35"),
(@ENTRY,0,13,14,61,0,100,0,0,0,0,0,75,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Aura 'Feign Death'"),
(@ENTRY,0,14,15,61,0,100,0,0,0,0,0,90,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Set Flag Standstate Dead"),
(@ENTRY,0,15,0,61,0,100,0,0,0,0,0,82,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Initialize - Add Npc Flags Gossip & Questgiver"),
(@ENTRY,0,16,0,6,0,100,0,0,0,0,0,6,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Just Died - Fail Quest 'Bitter Departure'"),
(@ENTRY,0,17,0,40,0,100,0,14,2943401,0,0,80,2943400,2,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Waypoint 14 Reached - Run Script");

-- Actionlist SAI
SET @ENTRY := 2943400;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,66,0,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Set Orientation Stored Target"),
(@ENTRY,9,1,0,0,0,100,0,1000,1000,0,0,1,1,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Say Line 1"),
(@ENTRY,9,2,0,0,0,100,0,0,0,0,0,15,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Quest Credit 'Bitter Departure'"),
(@ENTRY,9,3,0,0,0,100,0,5000,5000,0,0,69,0,0,0,0,0,0,8,0,0,0,6312.16,-1038.56,414.128,3.2475,"Injured Goblin Miner - On Script - Move To Position"),
(@ENTRY,9,4,0,0,0,100,0,5000,5000,0,0,41,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Despawn Instant");


-- gossip option conditions
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=15 AND `SourceGroup`=9859 AND `SourceEntry`=0;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(15,9859,0,0,0,9,0,12832,0,0,0,0,0,"","Show Gossip Injured Goblin Miner only if player has quest 'Bitter Departure' Taken");

DELETE FROM `waypoints` WHERE `entry`=2943400;
INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES 
(2943400, 1, 6645.47, -1263.66, 396.938, 'Injured Goblin Miner'),
(2943400, 2, 6662.06, -1274.13, 397.319, 'Injured Goblin Miner'),
(2943400, 3, 6666.29, -1254.59, 396.11, 'Injured Goblin Miner'),
(2943400, 4, 6669.33, -1240.06, 397.789, 'Injured Goblin Miner'),
(2943400, 5, 6654.31, -1205.01, 399.823, 'Injured Goblin Miner'),
(2943400, 6, 6657.03, -1170.15, 398.755, 'Injured Goblin Miner'),
(2943400, 7, 6667.33, -1146.28, 399.014, 'Injured Goblin Miner'),
(2943400, 8, 6670.9, -1138.68, 397.125, 'Injured Goblin Miner'),
(2943400, 9, 6685.09, -1103.95, 396.828, 'Injured Goblin Miner'),
(2943400, 10, 6688.71, -1094.32, 395.019, 'Injured Goblin Miner'),
(2943400, 11, 6684.75, -1084.31, 396.916, 'Injured Goblin Miner'),
(2943400, 12, 6679.3, -1074.4, 404.284, 'Injured Goblin Miner'),
(2943400, 13, 6691.91, -1051.96, 407.983, 'Injured Goblin Miner'),
(2943400, 14, 6696.28, -1016.54, 414.823, 'Injured Goblin Miner'),
(2943400, 15, 6675.32, -1005.06, 414.844, 'Injured Goblin Miner'),
(2943400, 16, 6661.01, -1007.41, 414.946, 'Injured Goblin Miner'),
(2943400, 17, 6643.75, -1010.24, 420.508, 'Injured Goblin Miner'),
(2943400, 18, 6620.76, -1012.59, 424.378, 'Injured Goblin Miner'),
(2943400, 19, 6610.33, -996.992, 428.116, 'Injured Goblin Miner'),
(2943400, 20, 6581.3, -1005.9, 434.257, 'Injured Goblin Miner'),
(2943400, 21, 6570.74, -1013, 434.076, 'Injured Goblin Miner'),
(2943400, 22, 6551.68, -1012.38, 433.218, 'Injured Goblin Miner'),
(2943400, 23, 6530.83, -1024.99, 433.04, 'Injured Goblin Miner'),
(2943400, 24, 6510.05, -1031.23, 435.824, 'Injured Goblin Miner'),
(2943400, 25, 6491.5, -1032.46, 434.226, 'Injured Goblin Miner'),
(2943400, 26, 6475.58, -1023.13, 434.759, 'Injured Goblin Miner'),
(2943400, 27, 6451.81, -1025.43, 431.502, 'Injured Goblin Miner');

DELETE FROM `waypoints` WHERE `entry`=2943401;
INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES 
(2943401,1,6675.684,-1133.351,397.0984, 'Injured Goblin Miner'),
(2943401,2,6685.072,-1098.411,396.1642, 'Injured Goblin Miner'),
(2943401,3,6681.2,-1080.326,399.9615, 'Injured Goblin Miner'),
(2943401,4,6679.967,-1046.549,409.238, 'Injured Goblin Miner'),
(2943401,5,6674.312,-1034.813,412.6546, 'Injured Goblin Miner'),
(2943401,6,6656.498,-1017.064,415.3774, 'Injured Goblin Miner'),
(2943401,7,6633.972,-1014.241,424.0492, 'Injured Goblin Miner'),
(2943401,8,6595.558,-1017.468,429.5381, 'Injured Goblin Miner'),
(2943401,9,6564.813,-1027.838,433.8102, 'Injured Goblin Miner'),
(2943401,10,6524.622,-1039.275,433.2169, 'Injured Goblin Miner'),
(2943401,11,6499.231,-1042.115,432.5026, 'Injured Goblin Miner'),
(2943401,12,6469.883,-1031.083,434.3615, 'Injured Goblin Miner'),
(2943401,13,6441.251,-1026.213,430.2799, 'Injured Goblin Miner'),
(2943401,14,6437.511,-1025.804,429.1273, 'Injured Goblin Miner');

DELETE FROM `waypoints` WHERE `entry`=2943402;
INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES 
(2943402,1,6664.493,-1106.511,423.9608, 'Injured Goblin Miner'),
(2943402,2,6674.33,-1120.81,424.4858, 'Injured Goblin Miner'),
(2943402,3,6669.01,-1129.653,423.9612, 'Injured Goblin Miner'),
(2943402,4,6634.209,-1128.281,417.0604, 'Injured Goblin Miner'),
(2943402,5,6620.618,-1119.293,412.5555, 'Injured Goblin Miner'),
(2943402,6,6617.512,-1092.118,411.8932, 'Injured Goblin Miner'),
(2943402,7,6620.42,-1077.044,415.567, 'Injured Goblin Miner'),
(2943402,8,6628.931,-1050.362,415.3565, 'Injured Goblin Miner'),
(2943402,9,6623.998,-1030.46,423.5757, 'Injured Goblin Miner'),
(2943402,10,6613.661,-1021.642,427.4341, 'Injured Goblin Miner'),
(2943402,11,6604.476,-1021.027,428.391, 'Injured Goblin Miner'),
(2943402,12,6603.476,-1023.777,428.391, 'Injured Goblin Miner'),
(2943402,13,6565.581,-1026.838,433.7635, 'Injured Goblin Miner'),
(2943402,14,6525.857,-1036.689,433.0396, 'Injured Goblin Miner'),
(2943402,15,6497.649,-1040.301,432.9883, 'Injured Goblin Miner'),
(2943402,16,6483.933,-1039.602,433.0212, 'Injured Goblin Miner'),
(2943402,17,6440.578,-1042.148,428.8166, 'Injured Goblin Miner');
```

**Comment by Teridan-dal-wow** [comment]
Thank you for the script!
Question: Why is `SET @ENTRY := -5004921` with "-" infront?
Also found a small bug, once a player completes the quest (doesn't matter with which npc it was done) the goblin will despawns at the end of the path or goblin dies = quest fails, in both cases the 25 secs respawn timer is trigger and when the goblin spawns it doesn't look like he has `aura=51329` and `bytes1=7` even thought you set it up in the script?!?

**Comment by Rushor** [comment]
> Question: Why is SET @ENTRY := -5004921 with "-" infront?

the three creature needs to have an individual script, so we load a script for each guid. in order to make the core recognise that it should load a guid script we put a - in front of the guid (else it would try to load an entryscript for 5004921 which of course does not exist)

I know coping the same script over and over even if it's only needed to load different paths is pretty dump, but @robinsch is already trying to work on a solution to load guid scripts and entryscripts at the same time

i will inspect the issue now

**Comment by Rushor** [comment]
> in both cases the 25 secs respawn timer is trigger and when the goblin spawns it doesn't look like he has aura=51329 and bytes1=7 even thought you set it up in the script?!?

fixed that by replacing the event 37 (init_ai) with event 11 (on_respawn) (updated the script in previous post)

> the goblin will despawns at the end of the script or goblin dies = quest fails

that works fine on PTR. i think that's a tc issue only then

**Comment by Teridan-dal-wow** [comment]
> that works fine on PTR. i think that's a tc issue only then

Ingore what i said, i didn't explain it properly, anyway it works as it should.

> fixed that by replacing the event 37 (init_ai) with event 11 (on_respawn) 

Tested all 3 of them, works perfectly now, thanks.

**Comment by Rushor** [comment]
alright with the new script update from https://github.com/dalaranwow/dalaran-wow/issues/4448 we can reduce the amount of db code to this:

```sql
-- Bitter Departure
UPDATE `creature_template` SET `ScriptName`='' WHERE  `entry`=29434;
DELETE FROM `script_waypoint` WHERE  `entry`=29434;

SET @CGUID := 5004920;
DELETE FROM `creature` WHERE `guid` BETWEEN @CGUID+0 AND @CGUID+1;
INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`,  `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `VerifiedBuild`) VALUES
(@CGUID+0, 29434, 571, 0, 0, 1, 1, 0,  0, 6646.888, -1110.959, 427.3377, 3.228859, 25, 0, 0, 0, 0, 0, 0, 0, 0, 23360), -- Injured Goblin Miner (Area: -Unknown-) (Auras: 51329 - Feign Death)
(@CGUID+1, 29434, 571, 0, 0, 1, 1, 0,  0, 6690.494, -1167.411, 398.8149, 3.769911, 25, 0, 0, 0, 0, 0, 0, 0, 0, 23360); -- Injured Goblin Miner (Area: -Unknown-) (Auras: 51329 - Feign Death)

UPDATE `creature` SET `saiscriptflag`=1 WHERE  `guid` IN (202337,@CGUID+0,@CGUID+1);

-- Injured Goblin Miner SAI
SET @ENTRY := 29434;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
-- gossip select part
(@ENTRY,0,0,1,62,0,100,0,9859,0,0,0,2,113,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Set Faction 113"),
(@ENTRY,0,1,2,61,0,100,0,0,0,0,0,64,1,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Store Targetlist"),
(@ENTRY,0,2,3,61,0,100,0,0,0,0,0,83,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Npc Flags Gossip & Questgiver"),
-- remove it on gossip select too if player starts the quest after serverdown
(@ENTRY,0,3,4,61,0,100,0,0,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Aura 'Feign Death'"),
(@ENTRY,0,4,5,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Remove Flag Standstate Dead"),
(@ENTRY,0,5,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Close Gossip"),
-- quest start part
(@ENTRY,0,6,7,19,0,100,0,12832,0,0,0,28,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Aura 'Feign Death'"),
(@ENTRY,0,7,8,61,0,100,0,0,0,0,0,91,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Remove Flag Standstate Dead"),
(@ENTRY,0,8,9,61,0,100,0,0,0,0,0,1,0,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Say Line 0"),
(@ENTRY,0,9,0,61,0,100,0,0,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Quest 'Bitter Departure' Taken - Set Active On"),
-- creature spawn part
(@ENTRY,0,10,11,11,0,100,0,0,0,0,0,2,35,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Respawn - Set Faction 35"),
(@ENTRY,0,11,12,61,0,100,0,0,0,0,0,75,51329,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Respawn - Add Aura 'Feign Death'"),
(@ENTRY,0,12,13,61,0,100,0,0,0,0,0,90,7,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Respawn - Set Flag Standstate Dead"),
(@ENTRY,0,13,0,61,0,100,0,0,0,0,0,82,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Respawn - Add Npc Flags Gossip & Questgiver"),
(@ENTRY,0,14,0,6,0,100,0,0,0,0,0,6,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Just Died - Fail Quest 'Bitter Departure'");

-- Injured Goblin Miner SAI
SET @GUID := -202337;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@GUID;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@GUID AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@GUID,0,0,0,62,0,100,0,9859,0,0,0,53,1,2943400,0,0,0,1,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Start Waypoint"),
(@GUID,0,1,0,40,0,100,0,27,2943400,0,0,80,2943400,2,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Waypoint 27 Reached - Run Script");

-- Injured Goblin Miner SAI
SET @GUID := -5004920;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@GUID;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@GUID AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@GUID,0,0,0,62,0,100,0,9859,0,0,0,53,1,2943402,0,0,0,1,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Start Waypoint"),
(@GUID,0,1,0,40,0,100,0,17,2943402,0,0,80,2943400,2,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Waypoint 27 Reached - Run Script");

-- Injured Goblin Miner SAI
SET @GUID := -5004921;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@GUID;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@GUID AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@GUID,0,0,0,62,0,100,0,9859,0,0,0,53,1,2943401,0,0,0,1,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Gossip Option 0 Selected - Start Waypoint"),
(@GUID,0,1,0,40,0,100,0,14,2943401,0,0,80,2943400,2,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Waypoint 14 Reached - Run Script");

-- Actionlist SAI
SET @ENTRY := 2943400;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,66,0,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Set Orientation Stored Target"),
(@ENTRY,9,1,0,0,0,100,0,1000,1000,0,0,1,1,6000,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Say Line 1"),
(@ENTRY,9,2,0,0,0,100,0,0,0,0,0,15,12832,0,0,0,0,0,12,1,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Quest Credit 'Bitter Departure'"),
(@ENTRY,9,3,0,0,0,100,0,5000,5000,0,0,69,0,0,0,0,0,0,8,0,0,0,6312.16,-1038.56,414.128,3.2475,"Injured Goblin Miner - On Script - Move To Position"),
(@ENTRY,9,4,0,0,0,100,0,5000,5000,0,0,41,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Injured Goblin Miner - On Script - Despawn Instant");

-- gossip option conditions
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=15 AND `SourceGroup`=9859 AND `SourceEntry`=0;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(15,9859,0,0,0,9,0,12832,0,0,0,0,0,"","Show Gossip Injured Goblin Miner only if player has quest 'Bitter Departure' Taken");

DELETE FROM `waypoints` WHERE `entry` IN (2943400, 2943401, 2943402);
INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES 
(2943400, 1, 6645.47, -1263.66, 396.938, 'Injured Goblin Miner'),
(2943400, 2, 6662.06, -1274.13, 397.319, 'Injured Goblin Miner'),
(2943400, 3, 6666.29, -1254.59, 396.11, 'Injured Goblin Miner'),
(2943400, 4, 6669.33, -1240.06, 397.789, 'Injured Goblin Miner'),
(2943400, 5, 6654.31, -1205.01, 399.823, 'Injured Goblin Miner'),
(2943400, 6, 6657.03, -1170.15, 398.755, 'Injured Goblin Miner'),
(2943400, 7, 6667.33, -1146.28, 399.014, 'Injured Goblin Miner'),
(2943400, 8, 6670.9, -1138.68, 397.125, 'Injured Goblin Miner'),
(2943400, 9, 6685.09, -1103.95, 396.828, 'Injured Goblin Miner'),
(2943400, 10, 6688.71, -1094.32, 395.019, 'Injured Goblin Miner'),
(2943400, 11, 6684.75, -1084.31, 396.916, 'Injured Goblin Miner'),
(2943400, 12, 6679.3, -1074.4, 404.284, 'Injured Goblin Miner'),
(2943400, 13, 6691.91, -1051.96, 407.983, 'Injured Goblin Miner'),
(2943400, 14, 6696.28, -1016.54, 414.823, 'Injured Goblin Miner'),
(2943400, 15, 6675.32, -1005.06, 414.844, 'Injured Goblin Miner'),
(2943400, 16, 6661.01, -1007.41, 414.946, 'Injured Goblin Miner'),
(2943400, 17, 6643.75, -1010.24, 420.508, 'Injured Goblin Miner'),
(2943400, 18, 6620.76, -1012.59, 424.378, 'Injured Goblin Miner'),
(2943400, 19, 6610.33, -996.992, 428.116, 'Injured Goblin Miner'),
(2943400, 20, 6581.3, -1005.9, 434.257, 'Injured Goblin Miner'),
(2943400, 21, 6570.74, -1013, 434.076, 'Injured Goblin Miner'),
(2943400, 22, 6551.68, -1012.38, 433.218, 'Injured Goblin Miner'),
(2943400, 23, 6530.83, -1024.99, 433.04, 'Injured Goblin Miner'),
(2943400, 24, 6510.05, -1031.23, 435.824, 'Injured Goblin Miner'),
(2943400, 25, 6491.5, -1032.46, 434.226, 'Injured Goblin Miner'),
(2943400, 26, 6475.58, -1023.13, 434.759, 'Injured Goblin Miner'),
(2943400, 27, 6451.81, -1025.43, 431.502, 'Injured Goblin Miner'),
(2943401,1,6675.684,-1133.351,397.0984, 'Injured Goblin Miner'),
(2943401,2,6685.072,-1098.411,396.1642, 'Injured Goblin Miner'),
(2943401,3,6681.2,-1080.326,399.9615, 'Injured Goblin Miner'),
(2943401,4,6679.967,-1046.549,409.238, 'Injured Goblin Miner'),
(2943401,5,6674.312,-1034.813,412.6546, 'Injured Goblin Miner'),
(2943401,6,6656.498,-1017.064,415.3774, 'Injured Goblin Miner'),
(2943401,7,6633.972,-1014.241,424.0492, 'Injured Goblin Miner'),
(2943401,8,6595.558,-1017.468,429.5381, 'Injured Goblin Miner'),
(2943401,9,6564.813,-1027.838,433.8102, 'Injured Goblin Miner'),
(2943401,10,6524.622,-1039.275,433.2169, 'Injured Goblin Miner'),
(2943401,11,6499.231,-1042.115,432.5026, 'Injured Goblin Miner'),
(2943401,12,6469.883,-1031.083,434.3615, 'Injured Goblin Miner'),
(2943401,13,6441.251,-1026.213,430.2799, 'Injured Goblin Miner'),
(2943401,14,6437.511,-1025.804,429.1273, 'Injured Goblin Miner'),
(2943402,1,6664.493,-1106.511,423.9608, 'Injured Goblin Miner'),
(2943402,2,6674.33,-1120.81,424.4858, 'Injured Goblin Miner'),
(2943402,3,6669.01,-1129.653,423.9612, 'Injured Goblin Miner'),
(2943402,4,6634.209,-1128.281,417.0604, 'Injured Goblin Miner'),
(2943402,5,6620.618,-1119.293,412.5555, 'Injured Goblin Miner'),
(2943402,6,6617.512,-1092.118,411.8932, 'Injured Goblin Miner'),
(2943402,7,6620.42,-1077.044,415.567, 'Injured Goblin Miner'),
(2943402,8,6628.931,-1050.362,415.3565, 'Injured Goblin Miner'),
(2943402,9,6623.998,-1030.46,423.5757, 'Injured Goblin Miner'),
(2943402,10,6613.661,-1021.642,427.4341, 'Injured Goblin Miner'),
(2943402,11,6604.476,-1021.027,428.391, 'Injured Goblin Miner'),
(2943402,12,6603.476,-1023.777,428.391, 'Injured Goblin Miner'),
(2943402,13,6565.581,-1026.838,433.7635, 'Injured Goblin Miner'),
(2943402,14,6525.857,-1036.689,433.0396, 'Injured Goblin Miner'),
(2943402,15,6497.649,-1040.301,432.9883, 'Injured Goblin Miner'),
(2943402,16,6483.933,-1039.602,433.0212, 'Injured Goblin Miner'),
(2943402,17,6440.578,-1042.148,428.8166, 'Injured Goblin Miner');
```

no longer copypaste shit over and over for guids - yay!

**Comment by Rushor** [comment]
DB/Quest: Injured Goblin Miner - Bitter Departure  - Add missing spawns and update questscript

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4303
* commit: 13d0c9cc46a8370e48fb73adcf0ae0042c8c5f5a

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by Rushor** [comment]
alright @Teridan-dal-wow you can now also test and use that new scriptengine if you compile and update:

https://github.com/dalaranwow/TrinityCore/commit/f2380a1e95d036cbd5eea8b0ed0e4c3163c97a89 and https://github.com/dalaranwow/TrinityCore/commit/4f6e440be9cdef3a52f99094ae990a884fa3a1b4

**Comment by Rushor** [comment]
DB/Creature: Injured Goblin Miner - Increase speedrate to 1.4 at the end of the escortevent

* updates: https://github.com/dalaranwow/dalaran-wow/issues/4303
* commit: 4c620e2309fcbbafb1072a3f6cbaa92ccbb7ca6c

**Comment by Teridan-dal-wow** [comment]
Yes, its probably my fault, but i spended the past 3 hours reinstalling my tc because 1 of those patch created a problem in my db and world server didn't want to start, so i will trust you xD

**Comment by Rushor** [comment]
you can setup the dalaran-wow server and the tc server together, you only need to use different db names for character/auth/world :)
